### PR TITLE
Create Coverity Collector

### PR DESCRIPTION
**Pull request by fidelcoria-aa**
image

Collector for security issues from Coverity for the security tab of the quality widget. The security tab is configured with a Coverity Stream and its corresponding Coverity Project.

UI changes:
Order rows in security tab by severity
Show status icons for security tab

### CODE REVIEW

File: collectors/security/coverity/src/main/java/com/capitalone/dashboard/model/CoverityCollector.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	
+	public static final String NICE_NAME = "Coverity";
+	
+    private List<String> coverityServers = new ArrayList<>();
+
+    public List<String> getCoverityServers() {
+        return coverityServers;
+    }
+
+    public static CoverityCollector prototype(List<String> servers) {
```
Comment (nireeshT) [inline collectors/security/coverity/src/main/java/com/capitalone/dashboard/model/CoverityCollector.java]: @fidelcoria-aa  How is coverity project uniquely identified?  project is mapped to collector-item , but there are options/unique options specified. Refer SonarCollector.java

Comment (fidelcoria-aa) [reply collectors/security/coverity/src/main/java/com/capitalone/dashboard/model/CoverityCollector.java]: Mmm... Do you mean CoverityProject.java?

**Comment by fidelcoria-aa** [comment]
I overlooked my comment in the CoverityProject.java file while preparing the PR. Looking back I think the setup works (ie uniquely identifying a `Coverity Collector-Item` as a Coverity Project _and_ Coverity Stream together. 

Would you agree with renaming this class to CoverityProjectStream.java?

File: collectors/security/coverity/src/main/java/com/capitalone/dashboard/model/CoverityCollector.java (hunk 1 above)
Comment (nireeshT) [reply collectors/security/coverity/src/main/java/com/capitalone/dashboard/model/CoverityCollector.java]: @fidelcoria-aa  Nope, take sonar collector 
{
    "_id" : ObjectId("56cb5bde7fab7c548551410e"),
    "_class" : "com.capitalone.dashboard.model.Collector",
    "name" : "Sonar",
    "collectorType" : "CodeQuality",
    "enabled" : true,
    "online" : true,
    "errors" : [],
    "uniqueFields" : {
        "projectName" : "",
        "instanceUrl" : ""
    },
    "allFields" : {
        "projectName" : "",
        "projectId" : "",
        "instanceUrl" : ""
    },
    "lastExecuted" : NumberLong(1551810772348),
    "searchFields" : [ 
        "options.projectName", 
        "niceName"
    ],
    "properties" : {}
}

can you send collector document from Mongo for Coverity collector. I am looking of block in Coverity collector 

"uniqueFields" : {
        "projectName" : "",
        "instanceUrl" : ""
    },
    "allFields" : {
        "projectName" : "",
        "projectId" : "",
        "instanceUrl" : ""
    },

Comment (fidelcoria-aa) [reply collectors/security/coverity/src/main/java/com/capitalone/dashboard/model/CoverityCollector.java]: "uniqueFields" : {},
    "allFields" : {},
Oooooh :open_mouth:.  I think I understand what you are saying. `uniqueFields` and `allFields` contain the fields of the collector-item that identify a unique item and all fields, respectively. 

Is this just for documentation purposes? I assume the values in these objects are meant for documentation. I don't see these values used anywhere. Or should I use them?

**Comment by fidelcoria-aa** [comment]
@nireeshT could you take a look at my comment in your review and confirm. Also for the comment immediately above this one :point_up:.

And another concern: In the screen shot above you can see the rows for each severity level, but they are not clickable, i.e. to see more details. However, The OPENSOURCE tab has a pop-up for details and so does the TEST tab. But STATIC ANALYSIS does not have a details pop-up, just a link.

Is there a reason the security widget and STATIC ANALYSIS do not have the details page like the other two? 
There is a PR in progress for a Fortify collector which is in the same situation.

**Comment by rvema** [comment]
@fidelcoria-aa We are working on separating the projects to seperate repos, Can you move this new collector to `https://github.com/Hygieia` ? Please let me know if you have any issues w.r.t. this.

**Comment by fidelcoria-aa** [comment]
@rvema Understood. but have 2 questions.
1. will each collector have its own repo? or will there be a repo for each category of collectors? like build category has jenkins, bamboo etc collectors.
2. I do not have permission to create a repo in Hygieia Org. Will you add me as a collaborator to the org?

**Comment by fidelcoria-aa** [comment]
Hey! I see there was a repo created for coverity collector! I will raise the pull request as is to that repo. I guess we will repeat the review comments from here in the new repo.

**Comment by fidelcoria-aa** [comment]
Raising the pull request to the new repo means I have to fork the new repo and then make a fork for coverity changes and **then** create PR.... I don't have enough time for this now. I will come back in a couple weeks.

**Pull request closed without merging**